quinn-proto/src/connection.rs

```diff
@@ -955,10 +948,11 @@ impl Connection {
                             trace!(self.log, "handshake ongoing");
                             self.handshake_cleanup(mux);
                             self.write_tls();
-                            return Ok(State::Handshake(state::Handshake {
+                            self.state = State::Handshake(state::Handshake {
                                 token: None,
                                 ..state
```

Comment:
Instead of cloning `state` above, could we do something like `if let State::Handshake(ref mut x) = self.state { x.token = None; } else { unreachable!(); }` here?

Reply:
No, because `ref mut state` overlaps with the `&mut self` which is used all over the place -- this was where I ran into trouble in the first place.

Reply:
Ah, I see. We could add a similar assignment block outside of the if statement to set `rem_cid_set`, but maybe it's not worth the effort.